maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/MultiThreadedBuilder.java

```diff
@@ -129,14 +132,23 @@ private void multiThreadedProjectTaskSegmentBuild( ConcurrencyDependencyGraph an
                                                        Map<MavenProject, ProjectSegment> projectBuildList,
                                                        ThreadOutputMuxer muxer )
     {
+        // gather artifactIds which are not unique so that the respective thread names can be extended with the groupId
+        Set<String> duplicateArtifactIds = projectBuildList.keySet().stream()
+                .map( MavenProject::getArtifactId )
+                .collect( Collectors.groupingBy( Function.identity(), Collectors.counting() ) )
```

Comment:
We could also stop here, usage in `createBuildCallable()` would then be something like:
```java
final String threadNameSuffix = artifactIdFrequency.get( project.getArtifactId() ) > 1
        //...
```